src/synology_dsm/api/core/external_usb.py

```diff
@@ -0,0 +1,245 @@
+"""DSM external USB device data."""
+
+from __future__ import annotations
+
+from synology_dsm.api import SynoBaseApi
+from synology_dsm.helpers import SynoFormatHelper
+
+
+class SynoCoreExternalUSB(SynoBaseApi):
+    """Class for external USB storage devices."""
+
+    API_KEY = "SYNO.Core.ExternalDevice.Storage.USB"
+    REQUEST_DATA = {"additional": '["all"]'}
+
+    async def update(self):
+        """Updates external USB storage device data."""
+        raw_data = await self._dsm.post(self.API_KEY, "list", data=self.REQUEST_DATA)
+        if raw_data:
+            device_data = raw_data["data"]
+            for device in device_data["devices"]:
+                self._data[device["dev_id"]] = SynoCoreExternalUSBDevice(device)
+
+    # Root
+    @property
+    def get_devices(self):
+        """Gets all external USB storage devices."""
+        return self._data
+
+    # Device
+    @property
+    def device_ids(self):
```

Comment:
I'm not sure, if this function is necessary :thinking: 
the same can be get by using the [`keys()`](https://docs.python.org/3.10/library/stdtypes.html#dict.keys) method of the dict returned by `get_devices()`